programs/mango-v4/src/accounts_ix/openbook_v2_cancel_order.rs

```diff
@@ -0,0 +1,55 @@
+use anchor_lang::prelude::*;
+
+use crate::error::*;
+use crate::state::*;
+use openbook_v2::{
+    program::OpenbookV2,
+    state::{Market, OpenOrdersAccount},
+};
+
+#[derive(Accounts)]
+pub struct OpenbookV2CancelOrder<'info> {
+    #[account(
+        constraint = group.load()?.is_ix_enabled(IxGate::OpenbookV2CancelOrder) @ MangoError::IxIsDisabled,
+    )]
+    pub group: AccountLoader<'info, Group>,
+
+    #[account(
+        mut,
+        has_one = group,
+        constraint = account.load()?.is_operational() @ MangoError::AccountIsFrozen
+        // owner is checked at #1
+    )]
+    pub account: AccountLoader<'info, MangoAccountFixed>,
+
+    pub authority: Signer<'info>,
+
+    #[account(
+        mut,
+        constraint = open_orders.load()?.market == openbook_v2_market_external.key()
```

Comment:
All accounts structs: I don't think this is the actual constraint we need. It would still be possible to pass in any openorders account, even for another users. Instead of this constraint, there should be a check that verifies that this openorders account is listed on the mangoaccount.

On the serum3 account this typically is done in instruction code and these get a comment like
```
/// CHECK: Validated inline by checking against the pubkey stored in the account at #2
```
I feel that would be appropriate here as well.

Reply:
This constraint just checks if the openorders belongs to the same market right. So I guess passing openorders for another user should work.